assert/_diff.ts

```diff
@@ -151,37 +152,38 @@ export function diff<T>(A: T[], B: T[]): Array<DiffResult<T>> {
   }
 
   function createFP(
-    slide: FarthestPoint,
-    down: FarthestPoint,
+    slide: FarthestPoint | undefined,
+    down: FarthestPoint | undefined,
     k: number,
     M: number,
   ): FarthestPoint {
     if (slide && slide.y === -1 && down && down.y === -1) {
       return { y: 0, id: 0 };
     }
-    if (
-      (down && down.y === -1) ||
+    const isAdding = (down?.y === -1) ||
       k === M ||
-      (slide && slide.y) > (down && down.y) + 1
-    ) {
+      (slide?.y || 0) > (down?.y || 0) + 1;
+    if (slide && isAdding) {
       const prev = slide.id;
       ptr++;
       routes[ptr] = prev;
       routes[ptr + diffTypesPtrOffset] = ADDED;
       return { y: slide.y, id: ptr };
-    } else {
+    } else if (down && !isAdding) {
       const prev = down.id;
       ptr++;
       routes[ptr] = prev;
       routes[ptr + diffTypesPtrOffset] = REMOVED;
       return { y: down.y + 1, id: ptr };
+    } else {
+      throw new Error("Unexpected missing FarthestPoint");
     }
   }
 
   function snake<T>(
     k: number,
-    slide: FarthestPoint,
-    down: FarthestPoint,
+    slide: FarthestPoint | undefined,
+    down: FarthestPoint | undefined,
```

Comment:
```suggestion
    slide?: FarthestPoint,
    down?: FarthestPoint,
```

Reply:
Issues with this, see comment https://github.com/denoland/deno_std/pull/4045#issuecomment-1875372330